redis/cluster.py

```diff
@@ -1393,14 +1404,6 @@ def __repr__(self):
     def __eq__(self, obj):
         return isinstance(obj, ClusterNode) and obj.name == self.name
 
-    def __del__(self):
-        try:
-            if self.redis_connection is not None:
-                self.redis_connection.close()
-        except Exception:
-            # Ignore errors when closing the connection
-            pass
-
```

Comment:
This causes issues with the change I made to `_get_or_create_cluster_node `, to make a new `ClusterNode` (sharing the `redis_connection` of the old node), instead of mutating the old one in place.

`Redis.__del__` already closes the connection pool down when the object is dropped, so I think this code was always unnecessary anyhow.